iceoryx_platform/linux/include/iceoryx_platform/attributes.hpp

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) 2021 - 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_HOOFS_LINUX_PLATFORM_ATTRIBUTES_HPP
+#define IOX_HOOFS_LINUX_PLATFORM_ATTRIBUTES_HPP
+
+#define IOX_NO_DISCARD
```

Comment:
This should be: `#define IOX_NO_DISCARD [[nodiscard, gnu::warn_unused]]`

Reply:
That doesn't compile as C++14 is set for the Linux build: `error: use of the 'nodiscard' attribute is a C++17 extension`

Reply:
aah then you have to implement it with the ifdef macros which distinct between c++14 and 17